src/routes/generator/Toolbar.svelte

```diff
@@ -1,51 +1,120 @@
 <script lang="ts">
 	import { isMapModalOpen, userProvidedLink } from './stores';
+	import Save from '../../icons/Save.svg';
+	import Redo from '../../icons/Redo.svg';
 	let nameOfMap = '';
 </script>
 
 <div class="buttonContainer">
-	<input
-		type="text"
-		class="mapNameInput"
-		value={nameOfMap}
-		placeholder="Please provide a name for your map"
-	/>
-	<input
-		type="text"
-		class="mapNameInput"
-		style="top:70px;"
-		bind:value={$userProvidedLink}
-		placeholder="Please provide a link to your image"
-	/>
-	<div class="shapeBox">Shapes here</div>
-	<button class="generateButton" style="right:460px;">Save to maps</button>
-	<button class="generateButton" style="right:330px;">Restart</button>
-	<button class="generateButton" on:click={() => ($isMapModalOpen = true)}>Generate map</button>
-	<div class="userProfile">
-		<div class="userProfileHead" />
-		<div class="userProfileBody" />
+	<div class="inputWrap">
+		<p style="position:absolute;color:white;left:590px;top:5%;font-weight:bold;">Shapes</p>
+		<label style="color:white; font-size:0.9rem">Name of the map:</label>
+		<input
+			type="text"
+			class="mapNameInput"
+			value={nameOfMap}
+			placeholder="Provide a name for the map"
+		/>
+		<label style="color:white;font-size:0.9rem">Link to original image:</label>
+		<input
+			type="text"
+			class="mapNameInput"
+			bind:value={$userProvidedLink}
+			placeholder="Provide a link to the original image"
+		/>
+	</div>
+	<div class="shapeBox">
+		<div class="saveButton" style="background-color:transparent">
+			<div class="rectShape" />
+		</div>
+		<div class="saveButton" style="background-color:transparent">
+			<div class="rectShape" style="border-radius:100%" />
+		</div>
+	</div>
+	<div class="buttonBox">
+		<div class="saveButton">
+			<img src={Save} />
+		</div>
+		<div class="saveButton" style="padding:3%">
+			<img src={Redo} />
+		</div>
+		<button class="generateButton" on:click={() => ($isMapModalOpen = true)}>Generate</button>
+		<div class="userProfile">
+			<div class="userProfileHead" />
+			<div class="userProfileBody" />
+		</div>
 	</div>
 </div>
 
 <style>
-	.shapeBox {
-		height: 70px;
-		width: 320px;
-		border-radius: 20px;
-		position: absolute;
-		top: 10px;
-		color: white;
-		padding: 10px;
-		left: 320px;
+	.buttonContainer {
+		display: flex;
+		align-items: center;
+		flex-direction: row;
+		background-color: var(--navyblue);
+		border: var(--lightbeige) solid 1px;
+		height: 140px;
+		border-radius: 40px;
+		margin-left: 100px;
+		margin-right: 100px;
 	}
-	.mapNameInput {
+	.buttonBox {
+		width: 400px;
```

Comment:
This will not scale, try to avoid setting a hardcoded width here and use padding instead

Reply:
resolved